Realm/RLMObjectSchema.mm

```diff
@@ -213,7 +213,7 @@ + (NSArray *)propertiesForClass:(Class)objectClass isSwift:(bool)isSwiftClass {
                 property.optional = true;
             }
             if (auto type = RLMCoerceToNil(propertyType)) {
-                if (existing == NSNotFound) {
+                if (existing == NSNotFound && ![ignoredProperties containsObject:propertyName]) {
```

Comment:
Why did you choose to perform the check at this point? From my reading of this method, the natural point would be to do this check as the first thing inside the block we give to `-enumerateKeysAndObjectsUsingBlock:`, before we do any other work for this property. I believe the two locations will give the same behavior, but it seems confusing to do work related to this property before checking whether it is ignored.